userlist.ts

```diff
@@ -0,0 +1,303 @@
+import { UserItem, UserHtmlElements } from "./types";
+import { data } from "./data";
+import { User } from "./user";
+import { callElements } from "./app";
+import { validate } from "./validate";
+
+let button: HTMLElement = document.getElementById("changeColor");
+let table: HTMLElement = document.getElementById("tablebody");
+
+export class UserList {
+  "users": UserItem[] = [];
+
+  constructor(init?: UserItem[]) {
+    Object.assign(this.users, init);
+  }
+
+  loadData = () => {
+    let items: UserItem[] = [...this.users] as UserItem[];
+
+    let itemsToDisplay: string[] = this.makeLayout(items);
+
+    table.innerHTML = itemsToDisplay.join("");
+
+    button.innerText = "Refresh data";
+    button.onclick = this.loadData;
+
+    //adding event to add new user
+    let submitnewuserbtn: HTMLElement = document.getElementById(
+      "submitnewuser"
+    );
+    submitnewuserbtn.addEventListener("click", this.addUser);
+
+    //adding event to validate user input
+    let elements: object = callElements("new", "input_");
```

Comment:
no need of object

Reply:
I am using this for input as well as normal div elements. div elements does not have value attribute, so using an interface here is giving issue.